service/service.go

```diff
@@ -669,12 +669,81 @@ func (s *Service) QueryDocument(ctx context.Context, req *pb.DocumentRequest) (*
 		return nil, errNilRequest
 	}
 
-	doc := req.GetData()
-	if doc == nil {
-		log.Println("[ERROR] Nil request data")
-		return nil, errNilRequestData
+	queryParams := req.GetQueryParameters()
+	if queryParams == nil {
+		log.Println("[ERROR] Nil query arguments")
+		return nil, errNilQueryArgs
 	}
 
-	return nil, nil
+	log.Printf("[INFO] QueryParameters contains:\n %s\n", queryParams)
+
+	log.Println("[INFO] Connecting to mongodb://hwscmongodb")
+	client, err := mongo.NewClient(conf.DocumentDB.Reader)
+	if err != nil {
+		log.Printf("[ERROR] Creating MongoDB client: %s\n", err.Error())
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	if err := client.Connect(context.TODO()); err != nil {
+		log.Printf("[ERROR] Connecting MongoDB client: %s\n", err.Error())
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	collection := client.Database(conf.DocumentDB.Name).Collection(conf.DocumentDB.Collection)
+
+	pipeline, err := buildAggregatePipeline(queryParams)
+	cur, err := collection.Aggregate(context.Background(), pipeline)
+	if err != nil {
```

Comment:
I don't see line 694's `err` being used, should it be `_` instead? (idk)

Line 694 and 695 `err` isn't considered redeclaring right? (that's just a question I had)

Reply:
fixed